src/array/fixed_size_list.rs

```diff
@@ -246,6 +246,78 @@ where
     }
 }
 
+/// An iterator over `N` elements of the iterator at a time.
+pub struct FixedSizeArrayChunks<const N: usize, I: IntoIterator> {
+    /// An owned iterator
+    iter: I::IntoIter,
+}
+
+impl<const N: usize, I: IntoIterator> FixedSizeArrayChunks<N, I> {
+    /// Returns a new [`FixedSizeArrayChunks`]
+    fn new(iter: I::IntoIter) -> Self {
+        Self { iter }
+    }
+}
+
+impl<const N: usize, I: IntoIterator> Iterator for FixedSizeArrayChunks<N, I> {
+    type Item = [I::Item; N];
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.iter
+            .by_ref()
+            .take(N)
+            .collect::<Vec<_>>()
+            .try_into()
+            .ok()
```

Comment:
Can't we avoid the allocation by using https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#initializing-an-array-element-by-element?